lib/sles4sap_publiccloud.pm

```diff
@@ -608,4 +610,63 @@ sub delete_network_peering {
     }
 }
 
+=head2 get_hana_database_info
+
+    Parses "hdbsql" command output.
+    Returns 1 if hana database is online otherwise 0 offline.
+
+=over 1
+
+=item B<RESULT> - default "-1" representing empty value.
+
+=back
+=cut
+
+sub get_hana_database_info {
+    my ($self) = @_;
+    my $result = -1;
+    $self->is_hana_online();
+    my $output_cmd = $self->run_cmd(cmd => "hdbsql -u SYSTEM -p Linux1234 -i 00 'SELECT * FROM SYS.M_DATABASES;'", runas => get_required_var("SAP_SIDADM"));
+    my $hana_database_online = '"HDB","HDB-00","YES","ACTIVE","","","DEFAULT","?"';
+
+    $result = ($output_cmd =~ /$hana_database_online/ ? 1 : 0);
+    record_info("Hana database status", $self->run_cmd(cmd => "hdbsql -u SYSTEM -p Linux1234 -i 00 'SELECT * FROM SYS.M_DATABASES;'"));
+    return $result;
+}
+
+=head2 is_hana_database_offline
+
+    Check if hana DB is offline.
+    Returns if hana database is online with 1 and offline with 0
+
+=over 1
+
+=item B<TIMEOUT> - default 900
+
+=back
+=cut
+
+sub is_hana_database_offline {
+    my ($self, %args) = @_;
+    my $timeout = bmwqemu::scale_timeout($args{timeout} // 900);
+    my $start_time = time;
+    my $db_status = -1;
+    my $consecutive_passes = 0;
+    my $total_consecutive_passes = ($args{total_consecutive_passes} // 5);
+
+    while ($consecutive_passes < $total_consecutive_passes) {
+        $db_status = $self->get_hana_database_info();
+        if (time - $start_time > $timeout) {
+            record_info("Hana database after timeout", $self->run_cmd(cmd => "hdbsql -u SYSTEM -p Linux1234 -i 00 'SELECT * FROM SYS.M_DATABASES;'"));
+            die("Hana database still is online");
+        }
+        if ($db_status eq 0) {
```

Comment:
Two things here:

1. In Perl, `eq` is a string operator and `==` is a numeric one. Since `$self->get_hana_database_info()` is returning a number (see L632), then it's better to use `==` in this condition.
2. This condition as written means that if in the first check the database is offline, then it will exit from the loop ... i.e., it's not going to check for consecutive passes. Is that intentional?

Reply:
refactored, yes it is intentional